### PR TITLE
[SYCL][Doc] Mark spec for prefetch extension as supported

### PR DESCRIPTION
**Pull request by bader**
According to release notes the extension was implemented by
https://github.com/intel/llvm/commit/e7139b0f15d0b5e9ebd8ea36c3982eb60fc1d42f,
https://github.com/intel/llvm/commit/0229456e19d5cc1771eb4e5a0a8f2d07afa4103f
and
https://github.com/intel/llvm/commit/b5d69df53620a1c9d33f7575cd3e6dd1ade54508

### CODE REVIEW

**Comment by bader** [comment]
@KornevNikita, did I get it right that this extension is implemented?

**Comment by KornevNikita** [comment]
@bader should we have cts to mark it as supported?

**Comment by bader** [comment]
I believe we do not have such a requirement. @gmlueck?

**Comment by gmlueck** [comment]
> I believe we do not have such a requirement. @gmlueck?

No, it is sufficient to have e2e tests in our own repo, and it looks like we did this in #11834.

However, there are two problems with the spec change in this PR:

* This extension is "experimental" not "supported", so it's in the wrong directory.
* The Status section text should be updated to note that it is "experimental".  See the [template](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/template.asciidoc#status) for the right wording.

Also, I don't see any definition of the feature-test macro `SYCL_EXT_ONEAPI_PREFETCH`.

**Comment by bader** [comment]
> However, there are two problems with the spec change in this PR:
> 
> * This extension is "experimental" not "supported", so it's in the wrong directory.
> * The Status section text should be updated to note that it is "experimental".  See the [template](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/template.asciidoc#status) for the right wording.

Fixed.

> Also, I don't see any definition of the feature-test macro `SYCL_EXT_ONEAPI_PREFETCH`.

It's here -> https://github.com/intel/llvm/blob/sycl/sycl/source/feature_test.hpp.in#L100

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_prefetch.asciidoc [hunk 1]
```diff
@@ -48,12 +48,11 @@ This extension also depends on the following other SYCL extensions:
...
-feedback.  Interfaces defined in this specification may not be implemented yet
-or may be in a preliminary state.  The specification itself may also change in
-incompatible ways before it is finalized.  *Shipping software products should
-not rely on APIs defined in this specification.*
-
+This is an experimental extension specification, intended to provide early
+access to features and gather community feedback. Interfaces defined in this
+specification are implemented in DPC++, but they are not finalized and may
+change incompatibly in future versions of DPC++ without prior notice. Shipping
+software products should not rely on APIs defined in this specification.
```
Comment (gmlueck) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_prefetch.asciidoc:55]:
```suggestion
change incompatibly in future versions of DPC++ without prior notice. *Shipping
software products should not rely on APIs defined in this specification.*
```

Let's keep the emphasis on that last sentence.

**Review by gmlueck — COMMENTED** [review]
>> Also, I don't see any definition of the feature-test macro SYCL_EXT_ONEAPI_PREFETCH.
>
> It's here -> https://github.com/intel/llvm/blob/sycl/sycl/source/feature_test.hpp.in#L100

Oh, sorry.  I just searched the "sycl/include" directory.

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_prefetch.asciidoc (hunk 1 above)
Comment (bader) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_prefetch.asciidoc]: Sorry, copied from the rendered version. Fixed.

**Review by gmlueck — APPROVED** [review]

**Pull request merged**